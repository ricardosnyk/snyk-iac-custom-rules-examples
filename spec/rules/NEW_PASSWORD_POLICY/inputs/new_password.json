{
  "format": "",
  "format_version": "",
  "input_type": "cloud_scan",
  "environment_provider": "cloud",
  "resources": {
    "aws_iam_account_password_policy": {
      "PasswordPolicy": {
        "id": "PasswordPolicy",
        "resource_type": "aws_iam_account_password_policy",
        "namespace": "us-east-1",
        "meta": null,
        "attributes": {
          "allow_users_to_change_password": true,
          "expire_passwords": true,
          "hard_expiry": false,
          "id": "PasswordPolicy",
          "max_password_age": 3,
          "minimum_password_length": 8,
          "password_reuse_prevention": 0,
          "require_lowercase_characters": false,
          "require_numbers": false,
          "require_symbols": false,
          "require_uppercase_characters": false
        }
      }
    }
  },
  "scope": {
    "environment_id": [
      "3b37a615-97dd-466f-ba2e-91aac91a032a"
    ],
    "org_id": "34c8f01d-ae1a-423d-a205-6eff50be87a6",
    "resource_type": [
      "aws_iam_account_password_policy"
    ]
  }
}