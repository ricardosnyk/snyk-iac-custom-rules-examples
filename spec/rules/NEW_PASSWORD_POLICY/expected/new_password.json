[
  {
    "passed": false,
    "ignored": false,
    "resource_id": "PasswordPolicy",
    "resource_namespace": "us-east-1",
    "resource_type": "aws_iam_account_password_policy",
    "severity": "high",
    "resources": [
      {
        "id": "PasswordPolicy",
        "type": "aws_iam_account_password_policy",
        "namespace": "us-east-1",
        "attributes": [
          {
            "path": [
              "minimum_password_length"
            ]
          }
        ]
      }
    ]
  }
]